### PR TITLE
MINOR: Rename log dir UUIDs

### PR DESCRIPTION
**Pull request by soarez**
After a late discussion in the voting thread for KIP-858 we decided to improve the names for the designated reserved log directory UUID values.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by clolov — APPROVED** [review]
For easy discoverability, this is the link to the discussion - https://lists.apache.org/thread/38sbj42ssoksjq2p488vdk1992dw3j1t. The change makes sense to me, thank you for persevering with this KIP!

File: clients/src/main/java/org/apache/kafka/common/Uuid.java [hunk 1]
```diff
@@ -49,21 +49,21 @@ public class Uuid implements Comparable<Uuid> {
     /**
      * A UUID that is used to identify new or unknown dir assignments.
      */
-    public static final Uuid UNKNOWN_DIR = ZERO_UUID;
+    public static final Uuid UNASSIGNED_DIR = ZERO_UUID;
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/Uuid.java]: The `Uuid` class should not know about any of this. We should move this logic to the module where it makes sense.

Comment (soarez) [reply clients/src/main/java/org/apache/kafka/common/Uuid.java]: Ok. Directory IDs will be used from metadata, core, storage and tools. I've moved them to a separate class, but I'm not sure which module would be a better fit than clients for this. Do you have a suggestion?

Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/common/Uuid.java]: `server-common` probably.

Comment (soarez) [reply clients/src/main/java/org/apache/kafka/common/Uuid.java]: Thanks for the suggestion. I've moved the class to that module.

File: server-common/src/main/java/org/apache/kafka/common/DirectoryId.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    /**
+     * Static factory to generate a directory ID.
+     *
+     * This will not generate a reserved UUID (first 100), or one whose string representation starts with a dash ("-")
+     */
+    public static Uuid random() {
+        Uuid uuid = Uuid.randomUuid();
+        while (RESERVED.contains(uuid) || uuid.toString().startsWith("-")) {
```
Comment (dengziming) [inline server-common/src/main/java/org/apache/kafka/common/DirectoryId.java:65]: We don't need `uuid.toString().startsWith("-")` here, right?

**Review by dengziming — APPROVED** [review]
LGTM, I think it's OK to check directory it twice.

**Comment by soarez** [comment]
Thank you for reviewing and merging @dengziming 🙏

**Pull request merged**